validation/urn.go

```diff
@@ -1,11 +1,24 @@
 package validation
 
 import (
+	"fmt"
+	"reflect"
+
+	"github.com/fabbricadigitale/scimd/api/attr"
+	"github.com/fabbricadigitale/scimd/schemas"
 	validator "gopkg.in/go-playground/validator.v9"
 )
 
 var urn = func(fl validator.FieldLevel) bool {
-	// (todo)> implement
+	field := fl.Field()
 
-	return true
+	switch field.Kind() {
+	case reflect.String:
+		str := field.String()
+		if attr.Parse(str).Valid() != true {
+			panic(fmt.Sprintf("Invalid URN composition: %+v", field.Interface()))
+		}
+		return schemas.URIRegexp.MatchString(str)
```

Comment:
`attr.Parse` is intended for attribute path usage, not URI only.

Please use regex and add special case `!strings.HasPrefix(strings.ToLower(a.URI), InvalidURNPrefix` within the validator.